### PR TITLE
Add script and action for updating a coverage badge.

### PR DESCRIPTION
**Pull request by joxl**
Adds a GitHub Action and supporting scripts to render a new coverage badge (referenced in the `README.md`) every time the main branch is updated.

### CODE REVIEW

**Review by millerdev — APPROVED** [review]

**Review by AmitPhulera — APPROVED** [review]

**Comment by dannyroberts** [comment]
(looks like you can merge this now if you like)

**Comment by joxl** [comment]
> (looks like you can merge this now if you like)

@dannyroberts, I'm holding this merge until #1 is merged (this PR's destination is the `dev` branch) since there's no rush for this merge prior to that one.

**Pull request merged**
